cli/cmd/deploy/deploy.go

```diff
@@ -478,6 +475,39 @@ func createProjectFlow(ctx context.Context, client *adminclient.Client, req *adm
 	return res, err
 }
 
+func variablesFlow(ctx context.Context, projectPath, projectName string) error {
+	connectorList, err := rillv1beta.ExtractConnectors(ctx, projectPath)
+	if err != nil {
+		return fmt.Errorf("failed to extract connectors %w", err)
+	}
+
+	// collect all sources
+	srcs := make([]*connectors.Source, 0)
+	for _, c := range connectorList {
+		if !c.AnonymousAccess {
+			srcs = append(srcs, c.Sources...)
+		}
+	}
+	if len(srcs) == 0 {
+		return nil
+	}
+
+	fmt.Printf("Rill does not have access to the following data sources:\n\n")
+	for _, src := range srcs {
+		if _, ok := src.Properties["path"]; ok {
+			// print URL wherever applicable
+			fmt.Printf(" - %s\n", src.Properties["path"])
+		} else {
+			fmt.Printf(" - %s\n", src.Name)
+		}
+	}
+	fmt.Printf("\nRun `rill env configure --project %s` to provide access to the data store.\n", projectName)
+	warn := color.New(color.Bold).Add(color.FgYellow)
+	warn.Printf("The project will remain in an error state until this is done.\n\n")
```

Comment:
1. Maybe print it all in warn color
2. Merge last log into the first – "Could not ingest all sources. Rill requires credentials for the following sources:"
3. Then make the last log: "Run `rill env configure --project %s` to provide credentials."